plugins/tableselection/plugin.js

```diff
@@ -777,53 +777,21 @@
 	function pasteListener( evt ) {
 		var editor = evt.editor,
 			selection = editor.getSelection(),
-			ranges = selection.getRanges(),
-			table = ranges.length && ranges[ 0 ]._getTableElement( { table: 1 } );
-
-		// (#2945)
-		if ( table && table.hasAttribute( ignoredTableAttribute ) ) {
-			return;
-		}
-
-		var selectedCells = getSelectedCells( selection ),
+			selectedCells = getSelectedCells( selection ),
 			pastedTable = this.findTableInPastedContent( editor, evt.data.dataValue ),
 			boundarySelection = selection.isInTable( true ) && this.isBoundarySelection( selection ),
```

Comment:
Nice idea, and agree on `pastedTable`. But `boundarySelection` is necessary right after declarations in [`shouldUseCustomPaste()`](https://github.com/ckeditor/ckeditor4/pull/3561/files/b9e556ebf2a23f900ee6b40615f501ea6b2c79ff#diff-3b0ed41f606e1508a1301dfcc588611eR865) (to be renamed) and also [later in the listener itself](https://github.com/ckeditor/ckeditor4/pull/3561/files/b9e556ebf2a23f900ee6b40615f501ea6b2c79ff#diff-3b0ed41f606e1508a1301dfcc588611eR830), so I guess it should be left as is.

Reply:
Well, I didn't notice at first but `pastedTable` is also used in both `isCustomPaste()` and in the listener, so it has to be calculated in the main scope right at the beginning.

Reply:
That's bad, it doesn't look like function requires those values. This might generate hard to spot errors in the future as a function became really tied to its context. I would go with direct passing those arguments to the function.